cluster/cluster.go

```diff
@@ -119,30 +111,30 @@ func (c *Cluster) setClusterDefaults() {
 			c.Nodes[i].HostnameOverride = c.Nodes[i].Address
 		}
 	}
-	if len(c.Services.KubeAPI.ServiceClusterIPRange) == 0 {
-		c.Services.KubeAPI.ServiceClusterIPRange = DefaultServiceClusterIPRange
-	}
-	if len(c.Services.KubeController.ServiceClusterIPRange) == 0 {
-		c.Services.KubeController.ServiceClusterIPRange = DefaultServiceClusterIPRange
-	}
-	if len(c.Services.KubeController.ClusterCIDR) == 0 {
-		c.Services.KubeController.ClusterCIDR = DefaultClusterCIDR
-	}
-	if len(c.Services.Kubelet.ClusterDNSServer) == 0 {
-		c.Services.Kubelet.ClusterDNSServer = DefaultClusterDNSService
-	}
-	if len(c.Services.Kubelet.ClusterDomain) == 0 {
-		c.Services.Kubelet.ClusterDomain = DefaultClusterDomain
-	}
-	if len(c.Services.Kubelet.InfraContainerImage) == 0 {
-		c.Services.Kubelet.InfraContainerImage = DefaultInfraContainerImage
-	}
-	if len(c.Authentication.Strategy) == 0 {
-		c.Authentication.Strategy = DefaultAuthStrategy
-	}
-	if len(c.Network.Plugin) == 0 {
-		c.Network.Plugin = DefaultNetworkPlugin
-	}
+	setDefaultIfEmpty(&c.Services.KubeAPI.ServiceClusterIPRange, DefaultServiceClusterIPRange)
+	setDefaultIfEmpty(&c.Services.KubeController.ServiceClusterIPRange, DefaultServiceClusterIPRange)
+	setDefaultIfEmpty(&c.Services.KubeController.ClusterCIDR, DefaultClusterCIDR)
+	setDefaultIfEmpty(&c.Services.Kubelet.ClusterDNSServer, DefaultClusterDNSService)
+	setDefaultIfEmpty(&c.Services.Kubelet.ClusterDomain, DefaultClusterDomain)
+	setDefaultIfEmpty(&c.Services.Kubelet.InfraContainerImage, DefaultInfraContainerImage)
+	setDefaultIfEmpty(&c.Authentication.Strategy, DefaultAuthStrategy)
+
+	c.setClusterNetworkDefaults()
+	c.setClusterImageDefaults()
+}
+
+func (c *Cluster) setClusterImageDefaults() {
+	if c.RKEImages == nil {
+		// don't break if the user didn't define rke_images
+		c.RKEImages = make(map[string]string)
+	}
+	setDefaultIfEmptyMapValue(c.RKEImages, "alpine", DefaultAplineImage)
```

Comment:
Consider doing it in some kind of a loop, there is a lot of repeatable stuff. When you read the config, you can put all images in a map, then iterate over this map